src/lib/masq.js

```diff
@@ -153,11 +155,28 @@ class Masq {
       const dbName = `app-${profileId}-${app.id}`
       const db = openOrCreateDB(dbName)
       this.appsDBs[dbName] = db
-      db.on('ready', () => this._startReplicate(db))
+      db.on('ready', () => {
+        const discoveryKey = db.discoveryKey.toString('hex')
+        this.appsDBs[discoveryKey] = db
+        this._startReplicate(db)
+      })
     })
 
     const profile = await this.getAndDecrypt('/profile')
     this.setState(STATES.LOGGED)
+
+    // register current device if it does not exist
+    const device = await this.getDevice()
+    if (!device) {
```

Comment:
TODO: move this to a function. This should add the device once and for all, for each new device